clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -31,8 +30,8 @@ void SYCLInstallationDetector::addSYCLIncludeArgs(
 }
 
 // Unsupported options for SYCL device compilation.
-static ArrayRef<OptSpecifier> getUnsupportedOpts() {
-  return {
+static ArrayRef<options::ID> getUnsupportedOpts() {
```

Comment:
maybe it's a good idea to redesign this function as returning an `ArrayRef` doesn't give too much.